src/DotNetProjectFile.Analyzers/Extensions/System.Xml.Linq.XElement.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace System.Xml.Linq;
+
+internal static class XElementExtensions
+{
+    public static bool IsSelfClosing(this XElement element)
+    {
+        if (element.Value is not { Length: > 0 })
+        {
+            return element.ToString()[^2] == '/';
```

Comment:
there can be optional spaces after the `/`, so this is not the best solution. In my PR I use a property to check for this, I don't recall by heart and doing this from phone.

Reply:
After? As far as I know, there can only be optional spacing before?!

Reply:
Hm, it indeed appears there can not be an optional space. But using the property in place to check for this still seems like a safer bet.

Reply:
`IsEmptyElement` on the reader of the node.